src/UmikoBot.cpp

```diff
@@ -741,6 +709,41 @@ QList<Command> UmikoBot::GetAllCommands()
 	return result;
 }
 
+void UmikoBot::VerifyAndRunAdminCmd(Client& client, const Message& message, const Channel& channel, unsigned int requiredNumberOfArgs, const QStringList& args, bool argumentShouldBeANumber, std::function<void()> callback)
+{
+	::Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN, [args, &client, message, channel, requiredNumberOfArgs, argumentShouldBeANumber, callback](bool result)
+	{
+		if (!result)
+		{
+			client.createMessage(message.channelId(), "**You don't have permissions to use this command.**");
+			return;
+		}
+
+		// TODO(fkp): Maybe we should have a parameter specifying if we want exact/at least $x args.
+		// I just don't want to have too many parameters, it's already quite long
```

Comment:
Maybe an idea is to send a regex string that needs to be matched, otherwise it will say "Wrong usage of command" ?